demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -869,9 +869,8 @@ static int establishMqttSession( MQTTContext_t * pMqttContext,
     MQTTStatus_t mqttStatus;
     MQTTConnectInfo_t connectInfo;
 
-    /* Buffer for storing client ID with random number.
-     * Note: The "+ 1U" is for the NULL character.*/
-    char clientIdBuffer[ MAX_CLIENT_IDENTIFIER_LENGTH + 1u ];
+    /* Buffer for storing client ID with random number. */
+    char clientIdBuffer[ MAX_CLIENT_IDENTIFIER_LENGTH ];
```

Comment:
```suggestion
    char clientIdBuffer[ CLIENT_IDENTIFIER_BUFFER_SIZE ];
```

Reply:
Was changed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1233/commits/ba24c0887ff413356328e21eaae9f0b3e3958a30